scheduler/server/persist_settings.go

```diff
@@ -0,0 +1,76 @@
+package server
+
+import (
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Persistor interface for persisting scheduler settings and initializing the scheduler
+// from its persisted settings
+type Persistor interface {
+	PersistSettings(settings *PersistedSettings) error
+	LoadSettings() (*PersistedSettings, error)
+}
+
+// PersistedSettings the persisted scheduler settings structure for encoding/decoding as json
+type PersistedSettings struct {
+	ClassLoadPercents               map[string]int32  `json:"classLoadPercents"`
+	RequestorToClassMap             map[string]string `json:"requestorToClassMap"`
+	RebalanceMinimumDurationMinutes int               `json:"rebalanceMinimumDurationMinutes"`
+	RebalanceThreshold              int               `json:"rebalanceThreshold"`
+	Throttle                        int               `json:"throttle"`
+}
+
+// nopPersistor provides nop implementations of persist and load functions
+type nopPersistor struct{}
+
+func (p *nopPersistor) PersistSettings(settings *PersistedSettings) error {
+	return nil
+}
+
+func (p *nopPersistor) LoadSettings() (*PersistedSettings, error) {
+	return nil, nil
+}
+
+func (s *statefulScheduler) persistSettings() {
```

Comment:
Return an error here and in loadSettings()?

Reply:
adding it to persist, but not to loadSettings().  loadSettings() already prints a log message that default values were used, the call to loadSettings() would do nothing with the error response.